Tidb support

morgo opened the pull request:
Add version detection for TiDB (detected_server).
Added support for tidb_snapshot option.
Auto-set tidb_snapshot to 1 second ago when TiDB detected.
Added optimization to dump _tidb_row id when it exists.
+ added a .gitignore file.

Tested on TiDB 2.1 + MySQL 5.7.19 (to make sure detected_server matched both correctly).

morgo commented:
Just an FYI: I removed a feature that I had added, but we no longer recommend (`_tidb_row_id`).  The diff is a little cleaner now, and what is tidb related is pretty self-contained.

maxbube commented on mydumper.c:
```diff
@@ -2500,6 +2552,10 @@ void dump_schema_data(MYSQL *conn, char *database, char *table, char *filename)
 		}
+	} else if (detected_server == SERVER_TYPE_TIDB) {
+		if (!skip_tz) {
+			g_string_append(statement,"/*!40103 SET TIME_ZONE='+00:00' */;\n");
```
shouldn't this be `g_string_printf` ?

maxbube commented on mydumper.c:
```diff
@@ -2923,6 +2979,10 @@ guint64 dump_table_data(MYSQL * conn, FILE *file, char *database, char *table, c
 					}
+				} else if (detected_server == SERVER_TYPE_TIDB) {
+					if (!skip_tz) {
+						g_string_append(statement,"/*!40103 SET TIME_ZONE='+00:00' */;\n");
```
shouldn't this be `g_string_printf` ?

maxbube commented on mydumper.c:
```diff
@@ -1353,6 +1375,37 @@ void start_dump(MYSQL *conn)
+ 
+		}
+ 
+	// Need to set the @@tidb_snapshot for the master thread
```
can you fix indentation for this block?

morgo commented:
I have addressed the comments in https://github.com/maxbube/mydumper/pull/155/commits/5e612b0df60c3f107d362fc736f13fdbe4d48d9e

Thank you for the review!

ihipop commented:
and update on this PR?
